### PR TITLE
apitrace: Added package

### PR DESCRIPTION
**Pull request by laanwj**
Add `apitrace` and `eglretrace`, for GLES debugging and profiling.

_Usage for tracing:_

```
apitrace trace -a egl ./glquake
```

This will create a file glquake.trace in the current directory (by default).

_Usage for replaying:_

```
eglretrace glquake.trace
```

There is also functionality for cutting/manipulating traces, for example to isolate certain frames, see `apitrace trim --help`.
It should also be possible to copy the resulting trace files to a desktop for examination using the qapitrace GUI (not yet tried).

### CODE REVIEW

File: package/apitrace/apitrace.mk [hunk 1]
```diff
@@ -0,0 +1,45 @@
+#############################################################
+#
+# apitrace
+#
+#############################################################
+
+#APITRACE_VERSION = 4
+#APITRACE_SITE = http://downloads.sourceforge.net/project/apitrace/apitrace/apitrace-$(APITRACE_VERSION)
+APITRACE_VERSION = master
+APITRACE_SITE = https://github.com/apitrace/apitrace.git
```
Comment (pcercuei) [inline package/apitrace/apitrace.mk]: Using git://github.com/apitrace/apitrace.git remove the need for the following line.

File: package/apitrace/apitrace.mk [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+APITRACE_WRAPPER_LIBRARIES = wrappers/egltrace.so
+
+ifneq ($(BR2_PACKAGE_XLIB_LIBX11),y)
+APITRACE_CONF_OPT += -DCMAKE_C_FLAGS:STRING="$(TARGET_CFLAGS) -DMESA_EGL_NO_X11_HEADERS"
+APITRACE_CONF_OPT += -DCMAKE_CXX_FLAGS:STRING="$(TARGET_CFLAGS) -DMESA_EGL_NO_X11_HEADERS"
+endif
+
+APITRACE_DEPENDENCIES = popt binutils
+
+define APITRACE_INSTALL_TARGET_CMDS
```
Comment (pcercuei) [inline package/apitrace/apitrace.mk]: Why define install/uninstall commands for a Cmake-based package?
If you need to install extra files, you can do the following:

define APITRACE_INSTALL_FOO
do_something
endef

 APITRACE_POST_INSTALL_TARGET_HOOKS += APITRACE_INSTALL_FOO

File: package/apitrace/apitrace.mk [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+APITRACE_VERSION = master
+APITRACE_SITE = https://github.com/apitrace/apitrace.git
+APITRACE_SITE_METHOD = git
+
+APITRACE_CONF_OPT = 
+APITRACE_BINARIES = apitrace eglretrace
+APITRACE_WRAPPER_LIBRARIES = wrappers/egltrace.so
+
+ifneq ($(BR2_PACKAGE_XLIB_LIBX11),y)
+APITRACE_CONF_OPT += -DCMAKE_C_FLAGS:STRING="$(TARGET_CFLAGS) -DMESA_EGL_NO_X11_HEADERS"
```
Comment (pcercuei) [inline package/apitrace/apitrace.mk:12]: I see you use -DMESA_EGL_NO_X11_HEADERS, I believe that means you are going to compile with at least EGL. If so, you have to add a dependency towards Mesa (and more specifically EGL if that is possible), either your package or the regular one.

File: package/apitrace/apitrace.mk [hunk 4]
```diff
@@ -0,0 +1,45 @@
+#############################################################
+#
+# apitrace
+#
+#############################################################
+
+#APITRACE_VERSION = 4
+#APITRACE_SITE = http://downloads.sourceforge.net/project/apitrace/apitrace/apitrace-$(APITRACE_VERSION)
```
Comment (pcercuei) [inline package/apitrace/apitrace.mk]: Unless they are used as a reminder, those two comment lines could be removed.

Comment (laanwj) [inline package/apitrace/apitrace.mk]: Well, it depends on whether you want to use a stable version or the latest from git, that's why I left both options. If git is preferred I'll remove these lines.

File: package/apitrace/apitrace.mk (hunk 1 above)
Comment (laanwj) [inline package/apitrace/apitrace.mk]: ok

File: package/apitrace/apitrace.mk (hunk 3 above)
Comment (laanwj) [inline package/apitrace/apitrace.mk:12]: apitrace does not depend on any of the drivers for APIs it traces directly; it includes all the header files it needs, hence my changes in thirdparty/khronos/EGL/eglplatform.h in the apitrace-master-001-fbdev patch to support building without X11. So I think the dependencies are correct.

File: package/apitrace/apitrace.mk (hunk 2 above)
Comment (laanwj) [inline package/apitrace/apitrace.mk]: Does it do a cmake install by default? I didn't know, will remove these.

File: package/apitrace/apitrace.mk (hunk 4 above)
Comment (pcercuei) [inline package/apitrace/apitrace.mk]: For external programs that we don't control, it's generally better to use a fixed version; either use a release or use a specific tag / hash for GIT. This ensures things won't break for us from one day to the other.

**Comment by laanwj** [comment]
Ok updated pull and processed comments, cmake indeed installs automatically so the .mk is a lot shorter now.

**Pull request merged**